src/log.h

```diff
@@ -20,83 +20,166 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #ifndef LOG_HEADER
 #define LOG_HEADER
 
+#include <map>
+#include <queue>
 #include <string>
+#include <fstream>
+#include "threads.h"
 
-/*
-	Use this for logging everything.
 
-	If you need to explicitly print something, use dstream or cout or cerr.
-*/
+namespace logging {
+
+/* Use this for logging everything.
+ * If you need to explicitly print something, use cout or cerr.
```

Comment:
`rawstream` is part of the logging setup, and this implies that the list is of non-logging utilities (dstream used to be separate from the logging facilities in `log.h`).